plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeLocalConcurrentCreateTableTest.java

```diff
@@ -0,0 +1,201 @@
+                            })
+                            .build())
+                    .forEach(MoreFutures::getDone);
+
+            // Verify each table was created with correct data
+            assertThat(query("SELECT * FROM test_ctas_1")).matches("VALUES (1, 10)");
+            assertThat(query("SELECT * FROM test_ctas_2")).matches("VALUES (11, 20)");
+            assertThat(query("SELECT * FROM test_ctas_3")).matches("VALUES (21, 30)");
+
+            // Verify table histories
+            assertQuery(
+                    "SELECT version, operation, isolation_level, is_blind_append FROM \"test_ctas_1$history\"",
+                    "VALUES (0, 'CREATE TABLE AS SELECT', 'WriteSerializable', true)");
+            assertQuery(
+                    "SELECT version, operation, isolation_level, is_blind_append FROM \"test_ctas_2$history\"",
+                    "VALUES (0, 'CREATE TABLE AS SELECT', 'WriteSerializable', true)");
+            assertQuery(
+                    "SELECT version, operation, isolation_level, is_blind_append FROM \"test_ctas_3$history\"",
+                    "VALUES (0, 'CREATE TABLE AS SELECT', 'WriteSerializable', true)");
+        }
+        finally {
+            // Clean up
+            assertUpdate("DROP TABLE IF EXISTS test_ctas_1");
+            assertUpdate("DROP TABLE IF EXISTS test_ctas_2");
+            assertUpdate("DROP TABLE IF EXISTS test_ctas_3");
+            executor.shutdownNow();
+            assertThat(executor.awaitTermination(10, SECONDS)).isTrue();
+        }
+    }
+
+    @Test
+    public void testConcurrentCreateTableAsSelectSameTable()
+            throws Exception
+    {
+        testConcurrentCreateTableAsSelectSameTable(false);
+        testConcurrentCreateTableAsSelectSameTable(true);
+    }
+
+    private void testConcurrentCreateTableAsSelectSameTable(boolean partitioned)
+            throws Exception
+    {
+        int threads = 3;
+        CyclicBarrier barrier = new CyclicBarrier(threads);
+        ExecutorService executor = newFixedThreadPool(threads);
+        String tableName = "test_concurrent_ctas_" + randomNameSuffix();
+
+        try {
+            getQueryRunner().execute("create table stg_test as SELECT a, b, 20220101 as d FROM UNNEST(SEQUENCE(1, 9001), SEQUENCE(1, 9001)) AS t(a, b)");
+
+            String selectString = " as ((select stg1.a as a, stg1.b as b, stg1.d as part from stg_test stg1, stg_test stg2 where stg1.d=stg2.d) " +
+                    "union all (select stg1.a as a, stg1.b as b, stg1.d as part from stg_test stg1, stg_test stg2 where stg1.d=stg2.d))";
+
+            // Execute concurrent CTAS operations
+            executor.invokeAll(ImmutableList.<Callable<Void>>builder()
+                            .add(() -> {
+                                barrier.await(10, SECONDS);
+                                try {
+                                    getQueryRunner().execute("CREATE TABLE " + tableName +
```

Comment:
If queries are the same use one lambda expression similar to:
https://github.com/trinodb/trino/blob/63fe63a5c9ad2ba3680427737261bf0d7cf86fbb/testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java#L5127

Reply:
done